File: route53-export-json.py
#!/usr/bin/env python
import json

import boto3

# Can also use an ~/.aws/credentials file
client = boto3.client(
    'route53',
    aws_access_key_id = '<access_key>',
    aws_secret_access_key = '<secret_key>',
)

response = client.list_hosted_zones(
    MaxItems='100',
)

for zone in response["HostedZones"]:
    record_sets = client.list_resource_record_sets(
        HostedZoneId = zone["Id"],
    )
    print (json.dumps(record_sets['ResourceRecordSets'], indent = 4, sort_keys = True, default = str ))

File: route53-export-json.pl
V#!/usr/bin/env perl 

use strict;
use warnings;
use utf8;

use WebService::Amazon::Route53;
use JSON;

my $client = WebService::Amazon::Route53->new(id => 'KEY_ID',
                                              key => 'SECRET_KEY');

my $response = $client->list_hosted_zones(max_items => 100);

for my $zones ($response->{'hosted_zones'}) {
    for my $zone (@$zones) {
        my $zonedetails = $client->list_resource_record_sets(zone_id => $zone->{'id'});
        print JSON::XS->new->utf8->pretty->encode($zonedetails);
    }
}
